addons/mail/static/src/model/model_manager.js

```diff
@@ -34,19 +34,29 @@ class ModelManager {
          * The messaging env.
          */
         this.env = env;
+
+        //----------------------------------------------------------------------
+        // Various variables that are necessary to handle an update cycle. The
+        // goal of having an update cycle is to delay the execution of computes,
+        // life-cycle hooks and potential UI re-renders until the last possible
+        // moment, for performance reasons.
+        //----------------------------------------------------------------------
+
+        /**
+         * Set of records that have been created during the current update
+         * cycle. Useful to trigger _created.
+         */
+        this._createdRecords = new Set();
         /**
-         * Whether this is currently handling an "update after" on a record.
-         * Useful to determine if we should process computed/related fields.
+         * Whether any change happened during the current update cycle. Useful
+         * to know if rev number has to be incremented.
          */
-        this._isHandlingToUpdateAfters = false;
+        this._hasAnyChangeDuringCycle = false;
         /**
-         * Determine whether an update cycle is currently in progress.
-         * Useful to determine whether an update should initiate an update
-         * cycle or not. An update cycle basically prioritizes processing
-         * of all direct updates (i.e. explicit from `data`) before
-         * processing computes.
+         * Set of records that have been updated during the current update
+         * cycle. Useful to increment record rev numbers.
```

Comment:
Same remark as before. Would also not mention rev number of record, but instead go straight to the end-goal which is to notify components that observe these records to re-render.